### PR TITLE
fix(replica): replica execute multi as single

### PR DESCRIPTION
**Pull request by adiholden**
The bug:
exec opcode not sent to replica
The temporary fix:
execute multi commands on replica side once they are received without waiting for exec

### CODE REVIEW

**Comment by romange** [comment]
FYI, this should not got to v1.14-branch, instead we will open another branch "v1.14-replicafix" and we will merge it there on top of 1.14.2 tag

File: src/server/replica.cc [hunk 1]
```diff
@@ -809,18 +809,18 @@ void DflyShardReplica::StableSyncDflyReadFb(Context* cntx) {
       break;
 
     last_io_time_ = Proactor()->GetMonotonicTimeNs();
-
-    if (!tx_data->is_ping) {
+    if (tx_data->opcode == journal::Op::PING) {
```
Comment (romange) [inline src/server/replica.cc:812]: it's just for counting stats ?

File: src/server/replica.cc [hunk 2]
```diff
@@ -1195,23 +1191,7 @@ auto DflyShardReplica::TransactionReader::NextTxData(JournalReader* reader, Cont
...
-    DCHECK(res->txid > 0);
-
-    auto txid = res->txid;
-    auto& txdata = current_[txid];
-    if (txdata.AddEntry(std::move(res.value()))) {
-      auto out = std::move(txdata);
-      current_.erase(txid);
-      return out;
-    }
+    return TransactionData::FromSingle(std::move(res.value()));
```
Comment (romange) [inline src/server/replica.cc:1194]: what was removed? (I am not familiar with this code)

File: src/server/replica.cc (hunk 1 above)
Comment (adiholden) [reply src/server/replica.cc:812]: no we send ack from replica to master when we get ping

File: src/server/replica.cc (hunk 2 above)
Comment (adiholden) [reply src/server/replica.cc:1194]: we dont accumulate the multi commands untill we get the exec opcode. instead we just execute them when we get them

Comment (dranikpg) [reply src/server/replica.cc:1194]: all the code above for collecting journal entries because we don't collect them anymore

**Review by romange — APPROVED** [review]

**Comment by romange** [comment]
probably not in this form. maybe we can wrap it with the flag

**Pull request closed without merging**